draft-kazuho-httpbis-priority.md

```diff
@@ -233,18 +233,78 @@ set to `3` and the progressive parameter set to `1`.
 priority = urgency=3, progressive=?1
 ~~~
 
-# Merging Client- and Server-Driven Parameters {#merging}
-
-It is not always the case that the client has the best understanding of how the
-HTTP responses deserve to be prioritized.  For example, use of an HTML document
-might depend heavily on one of the inline images.  Existence of such
-dependencies is typically best known to the server.
-
-By using the "Priority" response header, a server can override the
-prioritization hints provided by the client.  When used, the parameters found
-in the response header field overrides those specified by the client.
-
-For example, when the client sends an HTTP request with
+# Prioritization Process and Lifecycle
+
+HTTP/2 alludes to a prioritization process and does not specify many details
+about it. This might go some way to explain the different perspectives about
+prioritization that arise from different operators and use cases. The design
+also mixes concerns about prioritization between requests, streams and
+connections.
+
+HTTP/2's frame-based prioritization places an emphasis on the client's role in
+determining the priority of a response. {{?RFC7540}} Section 5.3 describes
+one key characteristic:
+
+> Explicitly setting the priority for a stream is input to a
+  prioritization process.  It does not guarantee any particular
+  processing or transmission order for the stream relative to any other
+  stream.  An endpoint cannot force a peer to process concurrent
+  streams in a particular order using priority.  Expressing priority is
+  therefore only a suggestion.
+
+In practice, this approach is quite one-sided and puts a burden on the endpoint
+with the most limited amount of information. A client processing HTML and
+generating requests generally has only a URL and a type (determined by an HTML
+element or other mechanism) on which to make a prioritization prediction.
+
+A server typically has more information about the resource than the client,
+including the context around it (such as linked representations). It uses local
```

Comment:
To me it seems that the comparison regarding the information that the endpoints have is a bit misleading.

Type (how a resource is used) is something that the client would always have, but something that a server "might" have. OTOH, it is true that the representation is only known to the server.

Maybe it's better to clarify that letting the server decide how to prioritize leads to better results, because the information that the both endpoints have can be "combined".

Reply:
Yes, I also don't really agree with how you're explaining it here.

While it's correct in principle, in practice the server typically doesn't really have that extended context available in an easy to use way or at least doesn't make appropriate use of it in most current deployments. Next to that, clients have more information than just URL and type, e.g., where the resource is in the HTML plays a large role, as well as how the internal implementation works.